src/Anonymization/Anonymizator.php

```diff
@@ -142,69 +167,167 @@ public function anonymize(
             }
 
             try {
-                yield \sprintf(' * table %d/%d: "%s" ("%s")', $count, $total, $table, \implode('", "', $targets));
-                yield "   - initializing anonymizers...";
+                $this->output->writeLine(\sprintf(
+                    ' * table %d/%d: "%s" ("%s")',
+                    $count,
+                    $total,
+                    $table,
+                    \implode('", "', $targets)
+                ));
+
+                $this->output->indent();
+                $this->output->write("- initializing anonymizers...");
+
                 \array_walk($anonymizers, fn (AbstractAnonymizer $anonymizer) => $anonymizer->initialize());
-                yield $this->printTimer($initTimer);
+
+                $timer = $this->formatTimer($initTimer);
+                $this->output->writeLine(\sprintf('[%s]', $timer));
 
                 $this->addAnonymizerIdColumn($table);
 
                 if ($atOnce) {
-                    yield from $this->anonymizeTableAtOnce($table, $anonymizers);
-                } else {
-                    yield from $this->anonymizeTablePerColumn($table, $anonymizers);
+                    $this->anonymizeTableAtOnce($table, $anonymizers);
+                    $this->logger->info(
+                        'Table "{table}" anonymized at once. Targets were: "{targets}" ({timer}).',
+                        $context + ['timer' => $timer]
+                    );
+                }
+                else {
+                    $this->anonymizeTablePerColumn($table, $anonymizers);
                 }
-            } finally {
+            }
+            catch (\Throwable $e) {
+                $this->logger->error(
+                    'Exception caught when anonymizing "{table}" table: {error}. (Targets were: "{targets}").',
+                    $context + ['error' => $e->getMessage()]
+                );
+            }
+            finally {
                 $cleanTimer = $this->startTimer();
-                // Cleanup everything, even in case of any error.
-                yield "   - cleaning anonymizers...";
+                // Clean up everything, even in case of any error.
+                $this->output->write("- cleaning anonymizers...");
                 \array_walk($anonymizers, fn (AbstractAnonymizer $anonymizer) => $anonymizer->clean());
 
                 $this->removeAnonymizerIdColumn($table);
 
-                yield $this->printTimer($cleanTimer);
-                yield '   - total ' . $this->printTimer($initTimer);
+                $cleanTimer = $this->formatTimer($cleanTimer);
+                $this->output->writeLine(\sprintf('[%s]', $cleanTimer));
+                $this->output->writeLine("- total " . $this->formatTimer($initTimer));
```

Comment:
I'm thinking that timer could be reused, it could be its own class instead, which starts when `new Timer()` and which outputs the result when calling `__toString()`.

Reply:
I agree, but that will be another PR :wink:

Reply:
#121